src/data/redux/grading/selectors.test.js

```diff
@@ -0,0 +1,833 @@
+    criteria: [
+      {
+        name: 'Ideas',
+        selectedOption: 'Fair',
+        feedback: feedbackRequirement.required,
+      },
+      {
+        name: 'Content',
+        selectedOption: 'Poor',
+        feedback: feedbackRequirement.optional,
+      },
+    ],
+  };
+  const testCriteriaInvalid = [
+    {
+      name: 'Ideas',
+      selectedOption: 'Fair',
+      feedback: 'some feedback for ideas',
+    },
+    {
+      name: 'Content',
+      selectedOption: '',
+      feedback: 'feedback for content',
+    },
+  ];
+  describe('validation.criteria selector', () => {
+    const { criteria } = selectors.validation;
+    describe('returned object', () => {
+      let cb;
+      beforeEach(() => {
+        cb = criteria.cb;
+      });
+      describe('feedback', () => {
+        const validFeedback = { feedback: 'Fair' };
+        const emptyFeedback = { feedback: '' };
+        const feedbackOptional = { feedback: feedbackRequirement.optional };
+        const feedbackRequired = { feedback: feedbackRequirement.required };
+        it('returns true iff feedback is required and not provided', () => {
+          const output = cb(
+            { criteria: [validFeedback, validFeedback, emptyFeedback] },
+            { criteria: [feedbackOptional, feedbackRequired, feedbackRequired] },
+          );
+          expect(output.map(({ feedback }) => feedback)).toEqual([true, true, false]);
+        });
+      });
+      describe('selectedOption', () => {
+        it('returns true iff selectedOption is empty', () => {
+          const output = cb(
+            { criteria: [{ selectedOption: 'Fair' }, { selectedOption: '' }] },
+            {},
+          );
+          expect(output.map(({ selectedOption }) => selectedOption)).toEqual([false, true]);
+        });
+      });
+    });
+  });
+  describe('validation.criterion selector', () => {
+    const testLocalState = { some: 'state' };
+    const testVal = 'my Test value!!!';
+    const testOrderNum = 'myORDERnum';
+    let criteria;
+    beforeAll(() => {
+      criteria = selectors.validation.criteria;
+      selectors.validation.criteria = (state) => ({ [testOrderNum]: { state, testVal } });
+    });
+    afterAll(() => {
+      selectors.validation.criteria = criteria;
+    });
+    it('returns the <orderNum> key value from the criteria selector data', () => {
+      expect(
+        selectors.validation.criterion(testLocalState, { orderNum: testOrderNum }),
+      ).toEqual({ state: testLocalState, testVal });
+    });
+  });
+  describe('validation.criterionFeedback', () => {
+    const orderNum = 'myORDERnum';
+    const testState = {
+      [orderNum]: { feedback: 'test feedback' },
+    };
+    let criterionFeedbackSelector;
+    beforeAll(() => {
+      criterionFeedbackSelector = selectors.validation.criterionFeedback;
+      // eslint-disable-next-line no-shadow
+      selectors.selected.criterionFeedback = (state, { orderNum }) => (state[orderNum].feedback);
```

Comment:
lint is actually right about this... If you are going to define orderNum in the upper scope, you can't use this trick here.
```suggestion
      selectors.selected.criterionFeedback = (state, params) => (state[params.orderNum].feedback);
```